src/coreclr/tools/Common/Compiler/TypeExtensions.cs

```diff
@@ -753,5 +753,41 @@ public static bool IsDynamicInterfaceCastableImplementation(this MetadataType in
             Debug.Assert(interfaceType.IsInterface);
             return interfaceType.HasCustomAttribute("System.Runtime.InteropServices", "DynamicInterfaceCastableImplementationAttribute");
         }
+
+        public static bool HasImpliedRepeatedFields(this MetadataType mdType)
+        {
+            if (mdType.IsInlineArray)
+            {
+                return true;
+            }
+
+            // If the type is not an [InlineArray] type, do a best-effort detection of whether the type is a fixed buffer type
+            // as emitted by the C# compiler.
+            FieldDesc firstField = null;
+            int numIntroducedFields = 0;
+            foreach (FieldDesc field in mdType.GetFields())
+            {
+                if (!field.IsStatic)
+                {
+                    firstField ??= field;
```

Comment:
Don't we just want `firstField = field;`. I don't see why we need the check at all.